store/v2alpha1/smt/store.go

```diff
@@ -77,6 +77,17 @@ func (s *Store) GetProofICS23(key []byte) (*ics23.CommitmentProof, error) {
 	return createIcs23Proof(s, key)
 }
 
+func (s *Store) GetSMTProof(key []byte) (*smt.SparseMerkleProof, error) {
+	if len(key) == 0 {
```

Comment:
The SMT seems to allow empty keys. The current implementation hashes the key to get a path. Is there a reason you're restricting this here? If you don't restrict it identically across the entire codebase, then that could lead to issues where an empty key is accepted in some places but not in others.

Reply:
Great point! I was going off of the implementation of `GetProof` written by the vulcanize team and added the check for consistency. `errKeyEmpty` is an error type included by them.

All the main methods that interact with the key/value pairs in `smt/store.go`, specifically `Set`, `Get`, `Delete`, and `Has` do not allow empty keys, so I think removing it here in `GetSMTProof` would make it inconsistent. The only way `cosmos-sdk` interacts with SMTs are through the `store` inside `smt/store.go` so I don't think there will cases where an empty key is accepted in some places and not in others.

Do you think it still makes sense to remove that check?

Reply:
Sounds good, if the check is everywhere then you could keep it.

The error should definitely be exported though. If not in this PR then open an issue so we don't forget, or put a TODO.